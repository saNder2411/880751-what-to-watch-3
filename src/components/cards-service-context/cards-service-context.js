import {createContext} from 'react';

const {
  Provider: CardsServiceProvider,
  Consumer: CardsServiceConsumer
} = createContext();

export {CardsServiceProvider, CardsServiceConsumer};
