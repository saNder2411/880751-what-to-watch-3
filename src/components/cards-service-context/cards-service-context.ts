import * as React from 'react';

const {
  Provider: CardsServiceProvider,
  Consumer: CardsServiceConsumer
} = React.createContext(void 0);

export {CardsServiceProvider, CardsServiceConsumer};
