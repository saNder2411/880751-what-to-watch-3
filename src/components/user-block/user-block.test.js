import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter, Switch, Route} from 'react-router-dom';
import CardsService from '../../services/cards-service.js';
import {CardsServiceProvider} from '../cards-service-context/cards-service-context.js';
import createAPI from '../../api.js';
import thunk from 'redux-thunk';

import UserBlock from './user-block.jsx';

const API = createAPI(() => {});
const cardsService = new CardsService(API);
const mockStore = configureStore([thunk]);

const mockUserDate = {
  id: 1,
  email: `vova@gmail.com`,
  name: `Vlad`,
  avatarSrc: `avatar_url`,
};


describe(`Render UserBlock`, () => {
  it(`Should UserBlock render correctly when user no authorization`, () => {
    const store = mockStore({
      user: {
        userData: {},
        authStatus: `NO_AUTH`,
        authLoading: false,
        authError: null,
      },
    });
    const markup = renderer
      .create(
          <Provider store={store}>
            <CardsServiceProvider value={cardsService}>
              <BrowserRouter>
                <Switch>
                  <Route
                    path='/'
                    component={UserBlock}
                  />
                </Switch>
              </BrowserRouter>
            </CardsServiceProvider>
          </Provider>
      )
      .toJSON();

    expect(markup).toMatchSnapshot();
  });

  it(`Should UserBlock render correctly when user authorization`, () => {
    const store = mockStore({
      user: {
        userData: mockUserDate,
        authStatus: `AUTH`,
        authLoading: false,
        authError: null,
      },
    });
    const markup = renderer
      .create(
          <Provider store={store}>
            <CardsServiceProvider value={cardsService}>
              <BrowserRouter>
                <Switch>
                  <Route
                    path='/'
                    component={UserBlock}
                  />
                </Switch>
              </BrowserRouter>
            </CardsServiceProvider>
          </Provider>
      )
      .toJSON();

    expect(markup).toMatchSnapshot();
  });
});

