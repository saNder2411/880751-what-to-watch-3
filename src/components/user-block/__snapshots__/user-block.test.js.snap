// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UserBlock Should UserBlock render correctly when user authorization 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="https://htmlacademy-react-3.appspot.com/avatar_url"
      width="63"
    />
  </div>
</div>
`;

exports[`Render UserBlock Should UserBlock render correctly when user no authorization 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
